### DISCUSSION TITLE
how do we create for a whole CIDR block?

### DISCUSSION THREAD

**Discussion by encryptblockr** [FiloSottile/mkcert#434] [Q&A] [answered]
Will like to create a certificate for all ip addresses in a CIDR block
Is this possible and/or supported at the moment?

**Comment by FiloSottile** [comment, marked as answer]
That's not supported by X.509 certificates in general.

**Comment by j1elo** [comment]
I was going to ask something similar. So maybe instead I'll just add here.

What is exactly not supported by X.509 certificates? The ability to specify a complete block, or the ability to specify too many individual items?

I need to generate some certs for a tutorial application, thus I don't know the private IP address of the users that will run the tutorial. Is there a common solution for that problem?

I just thought of writing a script that would generate ALL possible private range IPv4 addresses, and passing them to the `mkcert` command:

```sh
mkcert \
    192.168.0.1 \
    192.168.0.2 \
    192.168.0.3 \
    ...
    192.168.0.254 \
    192.168.1.1 \
    192.168.1.2 \
    192.168.1.3 \
    ...
```

Would this hit any technical limitation?
